### PR TITLE
refactor(FilterBar): add features that comply with the fiori guidelines

### PR DESCRIPTION
**Pull request by Lukas742**
* New `FilterGroupItem` that replaces the `FilterItem` component
* Add deprecation notice to `FilterItem`

* The FilterBar receives a filters dialog with new features. Eg.:
  * Search for filters
  * Possibility to hide/show filters in FilterBar
  * Filters are sorted by `groupName` prop of `FilterGroupItem`

**Thank you for your contribution!** 👏

To get it merged faster, kindly review the checklist below:

## Pull Request Checklist
- [x] Reviewed the [Contributing Guidelines](https://github.com/SAP/ui5-webcomponents-react/blob/master/CONTRIBUTING.md)
    + Especially the [How to Contribute](https://github.com/SAP/ui5-webcomponents-react/blob/master/CONTRIBUTING.md#contribute-code) section 
- [x] [Correct commit message style](https://github.com/SAP/ui5-webcomponents-react/blob/master/docs/Guidelines.md#commit-message-style)

### CODE REVIEW

File: packages/main/src/webComponents/Popover/Popover.stories.tsx [hunk 1]
```diff
@@ -1,64 +0,0 @@
-import { action } from '@storybook/addon-actions';
```
Comment (MarcusNotheis) [inline packages/main/src/webComponents/Popover/Popover.stories.tsx:1]: Why is this file deleted?

File: packages/main/src/webComponents/Dialog/Dialog.stories.tsx [hunk 2]
```diff
@@ -1,41 +0,0 @@
-import { action } from '@storybook/addon-actions';
```
Comment (MarcusNotheis) [inline packages/main/src/webComponents/Dialog/Dialog.stories.tsx:1]: why is this file deleted?

File: packages/main/src/components/ThemeProvider/ThemingSupport.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { root as sap_belize } from '@sap-theming/theming-base-content/content/Base/baseLib/sap_belize/variables.json';
+import { root as sap_belize_hcb } from '@sap-theming/theming-base-content/content/Base/baseLib/sap_belize_hcb/variables.json';
+import { root as sap_belize_hcw } from '@sap-theming/theming-base-content/content/Base/baseLib/sap_belize_hcw/variables.json';
+import { root as sap_fiori_3 } from '@sap-theming/theming-base-content/content/Base/baseLib/sap_fiori_3/variables.json';
```
Comment (MarcusNotheis) [inline packages/main/src/components/ThemeProvider/ThemingSupport.ts:4]: the fiori_3 theme can be removed from this map, it is included in the ThemeProvider already.
Looks like a merge issue?

File: packages/main/src/components/FilterItem/index.tsx [hunk 4]
```diff
@@ -49,6 +50,13 @@ const FilterItem: FC<FilterItemPropTypes> = forwardRef((props: FilterItemPropTyp
   } = props as FilterItemPropTypes;
   const classes = useStyles();
 
+  useEffect(() => {
+    deprecationNotice(
+      'FilterItem',
+      "'@ui5/webcomponents-react/lib/FilterItem' is deprecated.\nPlease use '@ui5/webcomponents-react/lib/FilterGroupItem' instead."
```
Comment (MarcusNotheis) [inline packages/main/src/components/FilterItem/index.tsx:56]:
```suggestion
      "'@ui5/webcomponents-react/lib/FilterItem' is deprecated and will be removed in the next major release.\nPlease use '@ui5/webcomponents-react/lib/FilterGroupItem' instead."
```

File: packages/main/src/components/FilterGroupItem/index.tsx [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+  groupName?: string;
+  labelTooltip?: string;
+  loading?: boolean;
+  mandatory?: boolean;
+  visible?: boolean;
+  visibleInFilterBar?: boolean;
+  considerGroupName?: boolean;
+}
+
+export const FilterGroupItem = forwardRef((props: FilterGroupItemPropTypes, ref: RefObject<HTMLDivElement>) => {
```
Comment (MarcusNotheis) [inline packages/main/src/components/FilterGroupItem/index.tsx:25]:
```suggestion
export const FilterGroupItem: FC<FilterGroupItemPropTypes> = forwardRef((props: FilterGroupItemPropTypes, ref: RefObject<HTMLDivElement>) => {
```

File: packages/main/src/components/FilterGroupItem/index.tsx [hunk 6]
```diff
@@ -0,0 +1,77 @@
+import { usePassThroughHtmlProps } from '@ui5/webcomponents-react-base/lib/usePassThroughHtmlProps';
+import { createComponentStyles } from '@ui5/webcomponents-react-base/lib/createComponentStyles';
+import { BusyIndicator } from '@ui5/webcomponents-react/lib/BusyIndicator';
+import { BusyIndicatorSize } from '@ui5/webcomponents-react/lib/BusyIndicatorSize';
+import { FlexBox } from '@ui5/webcomponents-react/lib/FlexBox';
+import { Label } from '@ui5/webcomponents-react/lib/Label';
+import React, { forwardRef, ReactElement, RefObject } from 'react';
```
Comment (MarcusNotheis) [inline packages/main/src/components/FilterGroupItem/index.tsx:7]:
```suggestion
import React, { FC, forwardRef, ReactElement, RefObject } from 'react';
```

File: packages/main/src/components/FilterGroupItem/index.tsx [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+    labelTooltip,
+    mandatory,
+    visible,
+    visibleInFilterBar,
+    children,
+    style,
+    loading,
+    // @ts-ignore
+    inFB
+  } = props;
```
Comment (MarcusNotheis) [inline packages/main/src/components/FilterGroupItem/index.tsx:40]: what about our other standard props like `className`, `style`, `tooltip` and `slot`?

File: packages/main/src/components/FilterGroupItem/index.tsx [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+
+  const passThroughProps = usePassThroughHtmlProps(props);
+
+  if (!mandatory && (!visible || (inFB && !visibleInFilterBar))) return null;
+  return (
+    <div
+      ref={ref}
+      {...passThroughProps}
+      className={inFB ? classes.filterItem : classes.filterItemDialog}
+      style={inFB ? style : {}}
```
Comment (MarcusNotheis) [inline packages/main/src/components/FilterGroupItem/index.tsx:50]: you can create the empty object outside of this component and use the `stableEmptyObject` instead

File: packages/main/src/components/FilterGroupItem/index.tsx [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+    >
+      <div className={inFB ? classes.innerFilterItemContainer : classes.innerFilterItemContainerDialog}>
+        <FlexBox>
+          {mandatory && <span className={classes.mandatory}>*</span>}
```
Comment (MarcusNotheis) [inline packages/main/src/components/FilterGroupItem/index.tsx:54]: Do you need a custom mandatory field? The `Label` component provides an `required` prop.

File: packages/main/src/components/FilterGroupItem/index.tsx [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+
+const useStyles = createComponentStyles(styles, { name: 'FilterGroupItem' });
+
+export interface FilterGroupItemPropTypes extends CommonProps {
+  children: ReactElement;
+  label?: string;
+  groupName?: string;
+  labelTooltip?: string;
+  loading?: boolean;
+  mandatory?: boolean;
```
Comment (MarcusNotheis) [inline packages/main/src/components/FilterGroupItem/index.tsx:19]: naming: This prop is called `required` in the label component. Should we adapt the same name here?

File: packages/main/src/components/FilterGroupItem/FilterGroupItem.jss.ts [hunk 11]
```diff
@@ -0,0 +1,53 @@
+import { ThemingParameters } from '@ui5/webcomponents-react-base/lib/ThemingParameters';
+
+const styles = {
+  filterItem: {
+    width: '13rem',
+    height: 'fit-content',
```
Comment (MarcusNotheis) [inline packages/main/src/components/FilterGroupItem/FilterGroupItem.jss.ts:6]: I think this is not working in our good old friend IE11. Is there any other CSS height possible?

File: packages/main/src/components/FilterBar/FilterBar.test.tsx [hunk 12]
```diff
@@ -1,130 +1,294 @@
...
 import { Input } from '@ui5/webcomponents-react/lib/Input';
+import { MultiComboBox } from '@ui5/webcomponents-react/lib/MultiComboBox';
+import { Option } from '@ui5/webcomponents-react/lib/Option';
+import { Select } from '@ui5/webcomponents-react/lib/Select';
 import { Switch } from '@ui5/webcomponents-react/lib/Switch';
 import { VariantManagement } from '@ui5/webcomponents-react/lib/VariantManagement';
+import { mount } from 'enzyme';
 import React from 'react';
+import { act } from 'react-dom/test-utils';
+import { MultiComboBoxItem } from "../../webComponents/MultiComboBoxItem";
```
Comment (MarcusNotheis) [inline packages/main/src/components/FilterBar/FilterBar.test.tsx:17]: please use @ui5/ import

File: packages/main/src/components/FilterBar/FilterDialog.tsx [hunk 13]
```diff
@@ -0,0 +1,252 @@
...
+      handleDialogClose(e);
+    },
+    [handleDialogCancel]
+  );
+
+  const renderFooter = useCallback(() => {
+    return (
+      <Bar
+        design={BarDesign.Footer}
+        renderContentRight={() => (
```
Comment (MarcusNotheis) [inline packages/main/src/components/FilterBar/FilterDialog.tsx:119]: no inline arrows please

File: packages/main/src/components/FilterBar/FilterDialog.tsx [hunk 14]
```diff
@@ -0,0 +1,252 @@
...
+        );
+      });
+  }, [renderChildren, toggledFilters, handleCheckBoxChange]);
+
+  return (
+    <Dialog ref={dialogRef} onAfterClose={handleClose} header={renderHeader()} footer={renderFooter()}>
+      <div className={classes.dialog}>
+        {renderFBSearch && (
+          <div className={classes.fbSearch} ref={searchRef}>
+            <span />
```
Comment (MarcusNotheis) [inline packages/main/src/components/FilterBar/FilterDialog.tsx:244]: empty span?

Comment (Lukas742) [reply packages/main/src/components/FilterBar/FilterDialog.tsx:244]: This is used as a placeholder to give the search row the same grid layout as the filter rows in the dialog box.

**Comment by Lukas742** [comment]
Closed because of merge issues.
